src/PerfView/Computers/RealtimeAntimalwareComputer.cs

```diff
@@ -0,0 +1,164 @@
+            if (processIndex == ProcessIndex.Invalid)
+                return;
+
+            ThreadIndex threadIndex = data.Thread().ThreadIndex;
+            if (threadIndex == ThreadIndex.Invalid)
+                return;
+
+            // Get the process container.
+            Dictionary<ThreadIndex, FileScanOperation> processContainer = GetOrCreateProcessContainer(processIndex);
+
+            // Create a new file scan operation.
+            // This happens when the scan is requested inside the user process.
+            FileScanOperation scan = new FileScanOperation()
+            {
+                File = data.FileName,
+                RequestorStack = _stackSource.GetCallStack(data.CallStackIndex(), data)
+            };
+
+            processContainer[threadIndex] = scan;
+        }
+
+        internal void StartScan(StreamscanrequestStartArgs_V1TraceData data)
+        {
+            // Get the requesting user process based on the PID logged inside the engine.
+            TraceProcess process = _traceLog.Processes.GetProcess(data.PID, data.TimeStampRelativeMSec);
+            ProcessIndex processIndex = process.ProcessIndex;
+            if (processIndex == ProcessIndex.Invalid)
+                return;
+
+            // Get the file scan operation.
+            Dictionary<ThreadIndex, FileScanOperation> processContainer = GetOrCreateProcessContainer(processIndex);
+            FileScanOperation operation = processContainer.Values.Where(s => s.File.Equals(data.Path, System.StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+            if(operation != null)
+            {
+                operation.StartTimeRelativeMSec = data.TimeStampRelativeMSec;
+                _engineThreadToScanMap[(int)data.Thread().ThreadIndex] = operation;
+            }
+        }
+
+        internal void StopScan(StreamscanrequestStartArgs_V1TraceData data)
+        {
+            // Get the requesting user process based on the PID logged inside the engine.
+            TraceProcess process = _traceLog.Processes.GetProcess(data.PID, data.TimeStampRelativeMSec);
+            ProcessIndex processIndex = process.ProcessIndex;
+            if (processIndex == ProcessIndex.Invalid)
+                return;
+
+            // Get the file scan operation.
+            Dictionary<ThreadIndex, FileScanOperation> processContainer = GetOrCreateProcessContainer(processIndex);
+            FileScanOperation operation = processContainer.Values.Where(s => s.File.Equals(data.Path, System.StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+            if (operation != null)
+            {
+                operation.StopTimeRelativeMSec = data.TimeStampRelativeMSec;
+
+                // Create the stack.
+                var fileNodeIndex = _stackSource.Interner.FrameIntern($"File ({operation.File})");
+                var resultNodeIndex = _stackSource.Interner.FrameIntern($"Scan Result ({operation.Result})");
```

Comment:
Suggestion: You might want to add the Reason from the AMFilter_FileScan event between the File and Scan Result.